airflow/jobs.py

```diff
@@ -851,6 +851,19 @@ def create_dag_run(self, dag, session=None):
                         "Next run date based on tasks %s",
                         next_run_date
                     )
+                if next_run_date and next_run_date <= dag.following_schedule(timezone.utcnow()):
+                    # if it is the very first run, set last run to previous period. This will force subsequent runs to be
```

Comment:
nit: break the comments at the new sentence e.g.

```suggestion
                    # if it is the very first run, set last run to previous period
                    # This will force subsequent runs to be in the next future period
```